TheMET/TheMET/ViewControllers/CategoryViewController.swift

```diff
@@ -28,6 +28,7 @@ class CategoryViewController: UIViewController, UITableViewDelegate, UITableView
     
     private let loadingCategoryView = LoadingPlaceholderView.construstView(configuration: .categoryArtworksLoading)
     private let failedCategoryView = FailedPlaceholderView.constructView(configuration: .categoryFailed)
+    private let failedSearchView = FailedPlaceholderView.constructView(configuration: .searchArtsFailed)
```

Comment:
This is an over-engineering, You already have a failedCategoryView. Instead of creating a new view, just switch the content of the failedCategoryView when needed.